system/doc/design_principles/statem.xml

```diff
@@ -1912,9 +1912,14 @@ do_unlock() ->
       to implicitly postpone any events to the <c>locked</c> state.
     </p>
     <p>
-      A selective receive cannot be used from a <c>gen_statem</c>
+      A receive should not be used from a <c>gen_statem</c>
       behaviour (or from any <c>gen_*</c> behaviour),
       as the receive statement is within the <c>gen_*</c> engine itself.
+      Doing so without taking all cases into consideration could result
+      in system messages being discarded leading to unexpected behaviour.
+      Even if all cases are taken into consideration the recvieve would
+      still block system messages from arriving in a timely manor or
+      indefinitely if a timeout is not provided.
```

Comment:
```suggestion
      indefinitely if a timeout was not provided.
```

Reply:
For consistency with "would", in the same sentence (?)

Reply:
Sorry, @paulo-ferraz-oliveira  you'll have to another pass as I put up a second pass ;)